src/Common/Internal/Http/HttpClient.php

```diff
@@ -127,6 +129,8 @@ public function __construct(
 
         $this->_requestUrl = null;
         $this->_expectedStatusCodes = [];
+
+        $this->_config = array_merge($this->_config, $options);
```

Comment:
Could the provided options be validated before hydrating `$this->_config`? (by instance, with an options resolver). BTW, if they are intended to be an associative array, `+` operator could be used instead of calling `array_merge()`.